src/clr-angular/forms/select/select.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Directive, ElementRef, HostListener, OnInit, Optional, Renderer2, ViewContainerRef } from '@angular/core';
+import { NgControl } from '@angular/forms';
+
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { ControlClassService } from '../common/providers/control-class.service';
+import { NgControlService } from '../common/providers/ng-control.service';
+import { WrappedFormControl } from '../common/wrapped-control';
+import { ClrSelectContainer } from './select-container';
+
+@Directive({ selector: '[clrSelect]', host: { '[class.clr-select]': 'true' } })
+export class ClrSelect extends WrappedFormControl<ClrSelectContainer> implements OnInit {
+  constructor(
+    vcr: ViewContainerRef,
+    @Optional() private ngControlService: NgControlService,
+    @Optional() private ifErrorService: IfErrorService,
+    @Optional() private control: NgControl,
+    @Optional() controlClassService: ControlClassService,
+    renderer: Renderer2,
```

Comment:
Are we using Renderer2 in this class? And some of these properties such as `vcr`, `el` are not private.

Reply:
No Renderer2 is not used so I removed that, but `vcr` and `el` are not available outside of the constructor. Unless you define `public` `protected` or `private` on a constructor property, they are not stored on the class as properties.